File: BE/src/main/java/bmg/nutrition_assistant/utils/Validator.java
package bmg.nutrition_assistant.utils;

public class Validator {
    public static boolean isEmailValid(String email) {
        return email.matches("^[A-Za-z0-9+_.-]+@\\w+\\.[.A-Za-z]{2,3}$");
    }

    public static boolean isPasswordValid(String password) {
        return password.length() >= 8;
        // return password.matches("^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=])(?=\\S+$).{8,}$");
    }

    public static boolean isNameValid(String name) {
        return name != null && !name.isEmpty() && name.length() <= 30;
    }

    public static boolean isHeightValid(Integer height) {
        return height == null || (height >= 0 && height <= 300);
    }

    public static boolean isWeightValid(Integer weight) {
        return weight == null || (weight >= 0 && weight <= 300);
    }
}

File: packages/backend/src/validator.ts
// Ported from utils/Validator.java
export const Validator = {
  isEmailValid(email: string): boolean {
    return /^[A-Za-z0-9+_.-]+@\w+\.[.A-Za-z]{2,3}$/.test(email);
  },

  isPasswordValid(password: string): boolean {
    return password.length >= 8;
  },

  isNameValid(name: string | null | undefined): boolean {
    return name != null && name.length > 0 && name.length <= 30;
  },

  isHeightValid(height: number | null | undefined): boolean {
    return height == null || (height >= 0 && height <= 300);
  },

  isWeightValid(weight: number | null | undefined): boolean {
    return weight == null || (weight >= 0 && weight <= 300);
  },
};
